src/routes/cart.js

```diff
@@ -0,0 +1,48 @@
+const router = require('express').Router();
+
+const CartModel = require('../../models/Carts');
+
+const { isAuthenticated } = require('../middlewares/authentication');
+
+router.put('/add/:productId', [isAuthenticated], async (req, res, next) => {
+  const { productId } = req.params;
+
+  try {
+    const userFavs = await CartModel.findOne({ userId: req.user });
+
+    if (!userFavs) {
+      const result = await CartModel.create({
+        userId: req.user,
+        products: [productId]
+      });
+
+      return res.status(201).json({
+        success: true,
+        count: result.products.length,
+        data: { products: [] }
+      });
+    }
+
+    if (userFavs.products.includes(productId)) {
+      throw new Error('product already in favorite list');
+    }
+
+    const result = await CartModel.findOneAndUpdate(
+      { userId: req.user },
+      { $push: { products: productId } },
+      { new: true }
```

Comment:
How about using the current `userFavs` you have available from querying first thing in the try statement instead  of using `new: true`?

This way we prevent re-executing a query and you could return instead:

```
const products = [...userFavs.get('products'), productId]
```

Reply:
```
const router = require('express').Router();

const CartModel = require('../../models/Carts');

const { isAuthenticated } = require('../middlewares/authentication');

router.put('/add/:productId', [isAuthenticated], async (req, res, next) => {
  const { productId } = req.params;

  try {
    const userCart = await CartModel.findOne({ userId: req.user });

    if (!userCart) {
      const result = await CartModel.create({
        userId: req.user,
        products: [productId],
      });

      return res.status(201).json({
        success: true,
        count: result.products.length,
        data: { products: [] },
      });
    }

    const products = userCart.get('products');

    // TODO Change code to support adding multiple products
    if (products.includes(productId)) {
      throw new Error('product already in favorite list');
    }

    await CartModel.findOneAndUpdate(
      { userId: req.user },
      { $push: { products: productId } }
    );

    res.status(200).json({
      success: true,
      count: products.length + 1,
      data: { products: [...products, productId] },
    });
  } catch (error) {
    next(error);
  }
});

module.exports = router;
```

Reply:
esto ya esta ready ready super ready 🦄